docs/UpgradeGuide.md

```diff
@@ -171,7 +201,78 @@ end
+| `Datadog.tracer.active_span.context.sampling_priority = Datadog::Ext::Priority::USER_REJECT` | `Datadog::Tracing.reject!`                  | See [sampling](#1.0-trace-api-sampling) for details.                |
+| `Datadog.tracer.active_span.context`                                                         | Removed                                     | Consider `Datadog::Tracing.active_trace` instead.                   |
+| `Datadog.tracer.active_span`                                                                 | `Datadog::Tracing.active_span`              | See [trace state](#1.0-trace-api-trace-state) for details.          |
+| `Datadog.tracer.call_context`                                                                | Removed                                     | See [trace state](#1.0-trace-api-trace-state) for details.          |
+| `Datadog.tracer.configure(options)`                                                          | `Datadog.configure { \|c\| ... }`           | Use configuration API instead.                                      |
+| `Datadog.tracer.provider.context = context`                                                  | `Datadog::Tracing.continue_from!(digest)`   | See [distributed tracing](#1.0-trace-api-distributed) for details.  |
+| `Datadog.tracer.set_tags(tags)`                                                              | `Datadog.configure { \|c\| c.tags = tags }` |                                                                     |
+| `Datadog.tracer.shutdown!`                                                                   | `Datadog::Tracing.shutdown!`                |                                                                     |
+| `Datadog.tracer.start_span`                                                                  | `Datadog::Tracing.trace`                    | See [manual tracing](#1.0-trace-api-manual-tracing) for details.    |
+| `Datadog.tracer.trace`                                                                       | `Datadog::Tracing.trace`                    | See [manual tracing](#1.0-trace-api-manual-tracing) for details.    |
+
+
+Also check out the functions defined within `Datadog::Tracing` in our [public API](https://www.rubydoc.info/gems/ddtrace/) for more details on their usage.
+
+<h3 id="1.0-trace-api-removed-tracer">Removed `Datadog.tracer`</h3>
+
+Many of the functions accessed directly through `Datadog.tracer` have been moved to `Datadog::Tracing` instead.
+
+<h3 id="1.0-trace-api-removed-context">Removed access to `Datadog::Context`</h3>
+
+Direct usage of `Datadog::Context` has been removed. Previously, it was used to modify or access active trace state. Most use cases have been replaced by our public trace API.
+
+<h3 id="1.0-trace-api-manual-tracing">Manual tracing & trace model</h3>
 
 Manual tracing is now done through the public API.
 
```

Comment:
⬇️ I can't comment on the next line directly (github...) but I found it confusing, since it states that "the block would yield" where actually it's the `Tracer#trace` method that yields to the block, and not the reverse.

Reply:
I don't see the confusion here, as a block yields an argument, even as a function yields arguments to a block.

Reply:
I think you may be getting a bit hung up on the special usage of the word yield in Ruby, whereas [in normal English](https://dictionary.cambridge.org/us/dictionary/english/yield) it means "provide", which is how I was using it here. I've changed "yield" to "provide" which should hopefully clarify and resolve this.

Reply:
In my defense, if I can get confused by this, our customers can too ;)

Reply:
Yup, hence the change to "provide" instead.

Reply:
Hmm I've reviewed the change and I still think it's not as clear is it could be. It's now "the block would provide" and "the block provides", where it's actually the reverse: the block gets provided/called -- it's on the receiving end, not on the supplying end.